conan/tools/gnu/pkgconfig.py

```diff
@@ -0,0 +1,95 @@
+import subprocess
+
+from conan.tools.env import Environment
+from conans.errors import ConanException
+from conans.util.runners import check_output_runner
+
+
+class PkgConfig:
+
+    def __init__(self, conanfile, library, pkg_config_path=None):
+        """
+        :param library: library (package) name, such as libastral
+        """
+        self._conanfile = conanfile
+        self._library = library
+        self._info = {}
+        self._pkg_config_path = pkg_config_path
+        self._variables = None
+
+    def _parse_output(self, option):
+        executable = self._conanfile.conf["tools.gnu:pkg_config"] or "pkg-config"
+        command = [executable, '--' + option, self._library, '--print-errors']
+        try:
+            env = Environment()
+            if self._pkg_config_path:
+                env.prepend_path("PKG_CONFIG_PATH", self._pkg_config_path)
+            with env.vars(self._conanfile).apply():
+                return check_output_runner(command).strip()
+        except subprocess.CalledProcessError as e:
+            raise ConanException('pkg-config command %s failed with error: %s' % (command, e))
+
+    def _get_option(self, option):
+        if option not in self._info:
+            self._info[option] = self._parse_output(option)
+        return self._info[option]
+
+    @property
+    def includedirs(self):
+        return [include[2:] for include in self._get_option('cflags-only-I').split()]
+
+    @property
+    def cflags(self):
+        return [flag for flag in self._get_option('cflags-only-other').split()
+                if not flag.startswith("-D")]
+
+    @property
+    def defines(self):
+        return [flag for flag in self._get_option('cflags-only-other').split()
+                if flag.startswith("-D")]
+
+    @property
+    def libdirs(self):
+        return [lib[2:] for lib in self._get_option('libs-only-L').split()]
+
+    @property
+    def libs(self):
+        return [lib[2:] for lib in self._get_option('libs-only-l').split()]
+
+    @property
+    def linkflags(self):
+        return self._get_option('libs-only-other').split()
+
+    @property
+    def provides(self):
+        return self._get_option('print-provides')
+
+    @property
+    def version(self):
+        return self._get_option('modversion')
+
+    @property
+    def variables(self):
+        if self._variables is None:
+            variable_names = self._parse_output('print-variables').split()
+            self._variables = {}
+            for name in variable_names:
+                self._variables[name] = self._parse_output('variable=%s' % name)
+        return self._variables
```

Comment:
is it possible to modernise, so it works like `CppInfo.variables.prefix` instead of `CppInfo.variables["prefix"]`?

Reply:
The ``getattr`` accessors we have in other locations of the codebase are an actual nightmare to debug and maintain. Unless strictly necessary, I won't be adding this syntax to new Conan helpers, and keep the native Python dict [] access